### PR TITLE
Add useMutation

### PR DESCRIPTION
**Pull request by ryonryon**

### CODE REVIEW

File: src/hooks/useMutation.test.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+describe("useMutation()", () => {
```
Comment (ryonryon) [inline src/hooks/useMutation.test.ts:1]: @kei95 Sorry I was stack with writing test. I am going to work soon in another PR

File: src/components/Project/DnDColumnList.tsx [hunk 2]
```diff
@@ -124,6 +124,7 @@ export default function DnDColumnList({ projectColumns }: DnDColumnListProps): J
                       ref={provided.innerRef}
                       {...provided.draggableProps}
                       {...provided.dragHandleProps}
+                      tabIndex={-1}
```
Comment (ryonryon) [inline src/components/Project/DnDColumnList.tsx:127]: here it is preventing tab to column component. I know it is not relating to the PR. but it is small enough and I just added anyway.

File: src/hooks/useMutation.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+interface ErrorBody {
+  status: number;
+  statusText: string;
+  message: string;
+}
+
+export function useMutation<TResponse>(
+  url: string,
+  method: "POST" | "PUT" | "PATCH" | "DELETE",
```
Comment (kei95) [inline src/hooks/useMutation.ts:15]: I think this hook is pretty much duplicating` useFetch()`. I think it's better to merge them both. Could you remove `useEffect` from the hook(`useFetch`) and return `makeRequest()` like this hook does?
This way you can remove this custom hook and `useFetch()` will be more flexible, which will achieve the same as this hook tires to do. The term `useMutation` is not really common in REST API anyway😄

File: src/hooks/useMutation.test.ts (hunk 1 above)
Comment (kei95) [reply src/hooks/useMutation.test.ts:1]: Np! I think you can even remove this hook and test and tweak  `useFetch()`.

File: src/hooks/useMutation.ts (hunk 3 above)
Comment (ryonryon) [reply src/hooks/useMutation.ts:15]: @kei95 
I thought about that way, in that case we have to write fetch like below.

```
const {makeRequest, ...} = useFetch(FETCH_URL(id));
const project = await makeRequest();
```
Do you have any idea that we can write one line for fetch? Yes, useMutation is from GraphQL term... lol

Comment (kei95) [reply src/hooks/useMutation.ts:15]: As a record, `useFetch()` and this hooks ain't too duplicating each other. Thus we'll keep this custom hook for when we want to make requests on demand👍

File: src/hooks/useMutation.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+      // Set an error if the request ins't successful
+      if (response.status < 200 || response.status > 299) {
+        const res = await response.blob();
+        const message = await res.text();
+
+        setError({ status: response.status, statusText: response.statusText, message });
+        setIsLoading(false);
+
+        console.log({ status: response.status, statusText: response.statusText, message });
```
Comment (kei95) [inline src/hooks/useMutation.ts]: Let's remove this console.log

**Review by kei95 — APPROVED** [review]

**Pull request merged**